internal/destination/config.go

```diff
@@ -86,6 +90,10 @@ func (c *Config) parseFields(cfg map[string]string) error {
 			return fmt.Errorf("parse %q: %w", ConfigKeyRetryAttempts, err)
 		}
 
+		if retryAttempts == 0 {
+			retryAttempts = defaultRetryAttempts
+		}
```

Comment:
IMO we should also reject negative values.

Reply:
or negative == retry forever (like most of clients' logic) ?

Reply:
Works for me as well. Maybe I'd just use `-1` as the magic value and return a validation error if it's any other negative number, just to be a bit more strict and prevent users from configuring infinite retries by mistake. That said, I don't think we should allow negative values for the wait between retries, that should be either 0 (default) or positive.